### DISCUSSION TITLE
What is the best way to display hierarchical data in a table?

### DISCUSSION THREAD

**Discussion by AlexKotik** [h2oai/wave#2075] [Q&A] [answered]
So basically I want to display some data that can have let's say recursively collapsed rows in a table. File listing is a good example of it. At first it shows directories in the collaped form, but then the user can expand directories to see files and subdirectories in the expanded directory. Right now it seems that table only supports grouping, but not the hierarcy (so there is no recursive grouping - groups within groups possible). What is the best way to display such kind of data? Maybe I could redraw the table each time user clicks on the element that can be expanded and add the "go up" button or something.

Here is an example how this kind of thing might look like (from ag-grid component site).
![ag-grid screenshot](https://github.com/h2oai/wave/assets/24991384/97d13e66-734e-4a50-b3c1-1a7744b8b815)

**Comment by mturoci** [comment, marked as answer]
Hi @AlexKotik, you basically answered both your questions :)

> Right now it seems that table only supports grouping, but not the hierarcy (so there is no recursive grouping - groups within groups possible).

Yes, recursive groups are not supported currently. You can file a feature request though as it's something that can be useful to others as well.

> What is the best way to display such kind of data? Maybe I could redraw the table each time user clicks on the element that can be expanded and add the "go up" button or something.

The table has `groups` attribute, which allows you to control / update the groups as needed. See [this](https://wave.h2o.ai/docs/examples/table-groups/) and [this](https://wave.h2o.ai/docs/examples/table-pagination-groups/) example  to get an idea. Then, you need some sort of state about what is the current position in hierarchy. Have a look at [breadcrumbs](https://wave.h2o.ai/docs/widgets/content/breadcrumbs/) which might come in handy for you.

@srini-x IIRC you have built something similar for Drive. Could you maybe put up some piece of code if you have any?
